### ISSUE TITLE
How to implement grdients like paper :Binarized Neural Networks ?

### ISSUE THREAD

**Issue by chaowang1994** [cc-hpc-itwm/TensorQuant#3]
How to implement grdients like paper :[Binarized Neural Networks](https://arxiv.org/abs/1602.02830)?
![image](https://user-images.githubusercontent.com/36892165/40461128-790b0352-5f3c-11e8-9ea0-9be4f61ec694.png)
I notice you assume TensorQuant gradient is always 1, and if I want to implement grdients like paper :Binarized Neural Networks ,what should I do?
Thanks a lot.

**Comment by DominikFHG** [comment]
In TensorQuant/Quantize/QuantKernelWrapper.py and TensorQuant/Quantize/FixedPoint.py you will find the wrappers which register the gradients of the different quantization methods (@ops.RegisterGradient). In Tensorflow you can apply relational operators to tensors, but the result will be a boolean type tensor, so you will need to cast it back to its original type. Also, you might find [this link](https://www.tensorflow.org/extend/adding_an_op#implement_the_gradient_in_python) useful, explaining how to register gradients in Tensorflow.
